checkout/access-checkout-android/src/main/java/com/worldpay/access/checkout/api/discovery/ApiDiscoveryAsyncTask.kt

```diff
@@ -11,6 +11,15 @@ import com.worldpay.access.checkout.logging.LoggingUtils.debugLog
 import java.net.MalformedURLException
 import java.net.URL
 
+/**
+ * [ApiDiscoveryAsyncTask] is responsible for carrying out the async discovery of the API endpoint for a service.
+ * Returns an [AsyncTaskResult] via callback
```

Comment:
this doesn't return an AsyncTaskResult via the callback, instead the callback uses the result to determine the parameters of the callback. I would change this to something like:

```
[ApiDiscoveryAsyncTask] is responsible for discovering an API given the list of endpoints. On completion, the given callback will be called with the corresponding error or success response from the Discovery API.
```

Reply:
Updated